cadc-access-control-server/src/main/java/ca/nrc/cadc/ac/server/ldap/LdapUserPersistence.java

```diff
@@ -578,7 +670,30 @@ public void resetPassword(HttpPrincipal userID, String newPassword)
             conns.releaseConnections();
         }
     }
-
+    
+    private boolean checkIfGroupExists(final Group group, final LdapGroupDAO groupDAO) {
+    	boolean groupExists = false;
+        try {
+            Group tempGroup = groupDAO.getGroup(group.getID().getName(), false);
+            if (tempGroup != null) {
+                groupExists = true;
+            }
+        } catch (GroupNotFoundException ex) {
+            try {
+                groupDAO.getUserAssociatedGroup(group.getID().getName(), false);
```

Comment:
The userDN for the LDAP_UNIQUE_MEMBER is different depending on whether the group is associated with a user or a userRequest.

Maybe able to add a getAnyGroup() which includes the LDAP_NSACCOUNTLOCK attribute in the search and check if the search result contains the LDAP_NSACCOUNTLOCK attribute. If it does not, then we generate the userDN for a group associated with a user, otherwise we generate the userDN for a group associated with a userRequest.

Will look into it further.

Reply:
Implemented LdapGroupDAO.getAnyGroup().